src/form.js

```diff
@@ -145,29 +145,49 @@ define([
     }
 
     Form.prototype = {
-        addHashtag: function(hashtag, xpath) {
-            this.hashtagDictionary[hashtag] = xpath;
+        isValidHashtag: function(tag) {
+            return this.hashtagDictionary.hasOwnProperty(this.normalizeHashtag(tag));
+        },
+        addHashtag: function(hashtag, xpath, dontOverwrite) {
+            if (!dontOverwrite || !this.hashtagDictionary[hashtag]) {
+                this.hashtagDictionary[hashtag] = xpath;
+            }
         },
         removeHashtag: function(hashtag) {
             delete this.hashtagDictionary[hashtag];
         },
-        normalizeHashtag: function (xpath_) {
+        transform: function(input, transformFn) {
+            input = this.normalizeEscapedHashtag(input);
+            return escapedHashtags.transform(input, transformFn);
+        },
+        normalize: function (methodName, xpath) {
             // try catch is needed as workaround for having an itemset without
-            // the itemset plugin enabled
+            // the itemset plugin enabled and invalid xpaths
             try {
-                return xpath_ ? this.xpath.parse(xpath_).toHashtag() : xpath_;
+                return xpath ? this.xpath.parse(xpath)[methodName]() : xpath;
             } catch (err) {
-                return xpath_;
+                return xpath.startsWith('#invalid/xpath ') ? xpath.slice(15) : xpath;
             }
+         },
+        normalizeEscapedHashtag: function (xpath_) {
+            return this.normalize('toEscapedHashtag', xpath_);
         },
-        normalizeXPath: function (xpath_, xpathParser) {
-            // if it's not an xpath just return the original string
+        normalizeHashtag: function (xpath_) {
+            return this.normalize('toHashtag', xpath_);
+        },
+        normalizeXPath: function (xpath_) {
+            return this.normalize('toXPath', xpath_);
+        },
+        hashtagsInXPath: function (xpath_) {
```

Comment:
Suggestion: rename to `getHashtagsInXPath`. At first I this would return a boolean value (true if there are hashtags in the xpath, false if not).